keymanapp/keyman
# fix(android/engine): Make it easier to test keyboard updates

Currently there's a lot of hoops involved for testing Keyman keyboard package updates (see https://github.com/keymanapp/keyman/pull/7740#issuecomment-1319640048)

This PR adds overrides for local and PR test builds to ignore calendar comparisons for checking keyboard updates. (Note: this won't apply to release builds used in Acceptance Testing)

I also separated the updateKMP comparison so that
* keyboard info is updated if the cloud has a newer version
* if updateKMP has a value, trigger notification for update available (this was causing updates to only appear on the 2nd boot)

Additional changes:
* Refactors version environment checks to new `VersionUtils()`

Commit https://github.com/keymanapp/keyman/commit/2d0008e2e38e1ffc0481af5f84a5f88bd8a1e3b7 is temporary test code for installing an old version 6.0.4 of sil_cameroon_qwerty.
- [x] DO NOT MERGE until reverting test code

## User Testing
Setup - Use Android device/emulator (phone in portrait orientation) with network access enabled

* **TEST_KEYBOARD_UPATES** 
1. Install the PR build of Keyman for Android
2. Launch Keyman for Android a first time.  This installs the old version 6.0.4 of sil_cameroon_qwerty
3. Completely close Keyman
4. Launch Keyman for Android a second time. This performs a check for keyboard updates
5. Dismiss the "Get Started" menu
6. Click the Keyman overflow menu
7. Verify icon for "Install Updates" is visible
